crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -93,10 +95,41 @@ pub enum Policy {
     Volume {
         /// Percentage of the order's volume should be taken as a protocol
         /// fee.
-        factor: f64,
+        factor: Factor,
     },
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Into)]
+pub struct Factor(f64);
+
+impl Factor {
+    /// Convert a partner fee into a `Factor` capping its value
+    pub fn partner_fee_capped_from(value: f64) -> Self {
+        Self(value.max(0.0).min(0.01))
```

Comment:
```suggestion
    pub fn partner_fee_capped_from(bps: f64) -> Self {
        Self(bps.max(0.0).min(0.01))
```

Reply:
@MartinquaXD what you propose is incorrect, it is parsing directly `f64` after the `bpd / 10_000`. We either keep it as it is or move the `bps` conversion inside this function (which I would avoid not to mix `bps` with `f64`.

Reply:
Ahh, true. Actually after thinking more about this the `Factor` type is very general purpose and the conversion from `bps` to a factor (**and capping that**) is very specific to the partner fee stuff and probably doesn't belong into the `Factor` implementation to begin with.
I mean conceptually a `Factor` can be used in lots of different contexts which are completely unrelated to fees.
WDYT?

Reply:
@MartinquaXD I agree with you, but in this case `Factor` is already particular for the fees (for example the fact that it cannot be `1.0` is very fee specific). I'd propose to rename it to `FeeFactor`, and then the capping function will have another input parameter to cap it with a max value (see https://github.com/cowprotocol/services/issues/2554). WDYT?